packages/browser/src/browser/index.ts

```diff
@@ -83,9 +83,16 @@ export function loadLegacySettings(
   const baseUrl = cdnURL ?? getCDN()
 
   return fetch(`${baseUrl}/v1/projects/${writeKey}/settings`)
-    .then((res) => res.json())
+    .then((res) => {
+      if (!res.ok) {
+        return res.text().then((errorResponseMessage) => {
+          throw { ...res, errorResponseMessage }
```

Comment:
We have metrics that keep track of errors when the library is loading, but it currently only works if the thing thrown is an error.

What do you think of throwing an error instead with the errorResponseMessage as the error.message?

Reply:
@cradek Really, the library only catches errors if the thing thing thrown passes an "instanceof Error" check? Seems dicyyy

Reply:
So just a quick update with: `throw {...res, message: errorResponseMessage }`? Works for me

Reply:
> @cradek Really, the library only reports errors if the thing thing thrown passes an "instanceof Error" check? Seems dicyyyyyy

Yeah - no we should definitely update that. We're also severely limited to how much text we can report (just learned about this last week), so we can't simply JSON.stringify whatever gets thrown and report it. 😢

Reply:
I also like how it looks better as well, great call!